src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -94,12 +94,33 @@ public Long getDurationInMinutes() {
         return Duration.between(initilizationDate, submissionDate).toMinutes();
     }
 
-    public Result getResult() {
-        return result;
+    /**
+     * Method used before changing relationship, first set to return the first in the ArrayList
+     * @return First Element in the ArrayList
+     */
+    public List<Result> getResults() {
+       return results;
     }
 
-    public void setResult(Result result) {
-        this.result = result;
+
+
+    /**
+     * Method use before changing relationship between result and submission, add one result to the List of the result
+     * @param results
+     */
+    public void setResults(List<Result> results) {
+        this.results = results;
+    }
+
+    public void addResult(Result result) {
```

Comment:
The implementation of this method is too complex. Do we really need this?
When we add results, we typically do not add the same one again.
The user of the Result class should have responsibility.
Removing existing results and adding new ones can lead to problems based on the cascade options and database constraints